upstream-sha256: 438f9dbd7ee2a025f36f869a3837fd72cdcd10d0ea2caec944a51bf082ba2ad8
actual-sha256: 8fb12fec76262a5b95c2b5bc72517be62c1af71346b270b0c96e0b19f6940a20
--- upstream.expected
+++ tolk.actual
@@ -18,7 +18,7 @@
   float val4 = (*(data4_16+(lidx1+4)));
   float val5 = (*(data4_16+(lidx1+8)));
   float4 val6 = (*((device float4*)((data3_16+alu0))));
-  *(data0_16+(lidx1+alu0)) = ((*(buf0+0))+(val6.x*val2)+(val6.y*val4)+(val6.z*val5)+(val6.w*val3));
+  *(data0_16+(lidx1+alu0)) = ((*(buf0+0))+(0.0f+(val6.x*val2)+(val6.y*val4)+(val6.z*val5)+(val6.w*val3)));
 }
 ---
 #include <metal_stdlib>
@@ -41,7 +41,7 @@
   float val4 = (*(data4_16+(lidx1+4)));
   float val5 = (*(data4_16+(lidx1+8)));
   float4 val6 = (*((device float4*)((data3_16+alu0))));
-  *(data0_16+(lidx1+alu0)) = (2.0f*((*(buf0+0))+(val6.x*val2)+(val6.y*val4)+(val6.z*val5)+(val6.w*val3)));
+  *(data0_16+(lidx1+alu0)) = (2.0f*((*(buf0+0))+(0.0f+(val6.x*val2)+(val6.y*val4)+(val6.z*val5)+(val6.w*val3))));
 }
 ---
 #include <metal_stdlib>
@@ -56,7 +56,7 @@
   float val0 = (*(data1_16+alu1));
   float4 val1 = (*((device float4*)((data2_16+alu0))));
   float4 val2 = (*((device float4*)((data3_16+(lidx1<<2)))));
-  *(data0_16+alu1) = ((2.0f*val0)+(val1.x*val2.x)+(val1.y*val2.y)+(val1.z*val2.z)+(val1.w*val2.w));
+  *(data0_16+alu1) = ((2.0f*val0)+(0.0f+(val1.x*val2.x)+(val1.y*val2.y)+(val1.z*val2.z)+(val1.w*val2.w)));
 }
 ---
 #include <metal_stdlib>
@@ -82,7 +82,7 @@
   float val7 = (*(data4_16+(lidx1+12)));
   float val8 = (*(data4_16+(lidx1+4)));
   float val9 = (*(data4_16+(lidx1+8)));
-  *(data0_16+(lidx1+(lidx0<<2))) = ((*(buf0+0))+(val2*val6)+(val4*val8)+(val5*val9)+(val3*val7));
+  *(data0_16+(lidx1+(lidx0<<2))) = ((*(buf0+0))+(0.0f+(val2*val6)+(val4*val8)+(val5*val9)+(val3*val7)));
 }
 ---
 #include <metal_stdlib>
@@ -108,7 +108,7 @@
   float val7 = (*(data4_16+(lidx1+12)));
   float val8 = (*(data4_16+(lidx1+4)));
   float val9 = (*(data4_16+(lidx1+8)));
-  *(data0_16+(lidx1+(lidx0<<2))) = ((*(buf0+0))+(val2*val6)+(val4*val8)+(val5*val9)+(val3*val7));
+  *(data0_16+(lidx1+(lidx0<<2))) = ((*(buf0+0))+(0.0f+(val2*val6)+(val4*val8)+(val5*val9)+(val3*val7)));
 }
 ---
 #include <metal_stdlib>
@@ -121,5 +121,5 @@
   float4 val0 = (*((device float4*)((data1_16+alu0))));
   int lidx1 = lid.y; /* 4 */
   float4 val1 = (*((device float4*)((data2_16+(lidx1<<2)))));
-  *(data0_16+(lidx1+alu0)) = ((val0.x*val1.x)+(val0.y*val1.y)+(val0.z*val1.z)+(val0.w*val1.w));
+  *(data0_16+(lidx1+alu0)) = (0.0f+(val0.x*val1.x)+(val0.y*val1.y)+(val0.z*val1.z)+(val0.w*val1.w));
 }
